mbed-drivers/v2/I2C.hpp

```diff
@@ -0,0 +1,605 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2006-2016 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_DRIVERS_V1_I2C_HPP
+#define MBED_DRIVERS_V1_I2C_HPP
+
+#include "mbed-drivers/platform.h"
+
+#if DEVICE_I2C && DEVICE_I2C_ASYNCH
+
+
+#include "mbed-hal/i2c_api.h"
+#include "mbed-hal/dma_api.h"
+
+#include "mbed-drivers/CThunk.h"
+#include "core-util/FunctionPointer.h"
+#include "core-util/PoolAllocator.h"
+
+// Forward declarations
+namespace mbed {
+namespace drivers {
+namespace v2 {
+enum class I2CError;
+} // namespace mbed
+} // namespace drivers
+} // namespace v2
+
+#include "I2CDetail.hpp"
```

Comment:
Why? It lives in the same directory. `#include "I2CDetail.hpp"` should search local directories first, which is precisely the behaviour we want here.